includes/ui/RedirectMessage.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Frontend redirect messages.
+ *
+ * @package wp-job-manager
+ * @since $$next-version$$
+ */
+
+namespace WP_Job_Manager\UI;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Redirect after an action is finished and make sure a message can displayed on the next page load.
+ *
+ * Stores the messages in a transient with an ID, and passes along the ID as a query string to the redirected URL.
+ *
+ * @since $$next-version$$
+ *
+ * @internal This API is still under development and subject to change.
+ * @access private
+ */
+class RedirectMessage {
+
+	const DEFAULT_QUERY_VAR = 'message';
+	const TRANSIENT_PREFIX  = 'wpjm_message_';
+
+	/**
+	 * Redirect to a URL with a message to be displayed on the next page load.
+	 *
+	 * @param string $url URL to redirect to.
+	 * @param string $message Optional message to be displayed on the next page load.
+	 * @param string $query_var A query variable added to the redirect URL, referencing the message transient.
+	 */
+	public static function redirect( $url, $message = null, $query_var = '' ) {
+		if ( ! empty( $message ) ) {
+			$url = add_query_arg( [ $query_var ?? self::DEFAULT_QUERY_VAR => self::set_transient_message( $message ) ], $url );
```

Comment:
The ?? operator will still return $query_var if it's empty, for a shorthand, you can use the short-ternary operator (?:), but I'd say that a proper if/else here might be better for readability

Reply:
Alternatively you can also set the default $query_var to null, but it might be better to use empty( $query_var ) anyway, so it covers all empty cases and never fatals

Reply:
Good catch, thanks. Fixing in 7059ce8